### PR TITLE
BPS-447 Update ocr data format to list of fields

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-447

### Change description ###
Updated ocr data format according to the processor changes.
Processor changes:  https://github.com/hmcts/bulk-scan-processor/pull/433

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java [hunk 1]
```diff
@@ -212,4 +215,18 @@ private JSONObject toJson(Document doc) throws Exception {
...
             .put("url", doc.url);
     }
+
+    private JSONArray toOcrJson(List<OcrDataField> ocrDataFields) throws JSONException {
+        JSONArray ocrJson = new JSONArray();
+        JSONObject ocrDataEntry;
+
+        for (OcrDataField ocrField : ocrDataFields) {
+            ocrDataEntry = new JSONObject();
+            ocrDataEntry.put("metadata_field_name", ocrField.name);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java]: Will this change the order ?

Comment (aliveni) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java]: No, this won't change the order

Comment (aliveni) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java]: We have a test in `OcrDataOrderTest` class, where we are testing the order of the fields.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/OcrDataField.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class OcrDataField {
+
+    @JsonProperty(value = "metadata_field_name")
+    public String name;
+    @JsonProperty(value = "metadata_field_value")
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/OcrDataField.java]: those are used for serialization, are they needed?

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/OcrDataField.java]: no, I removed them

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java [hunk 3]
```diff
@@ -9,6 +9,14 @@
...
     public int compare(CcdCollectionElement<CcdKeyValue> o1, CcdCollectionElement<CcdKeyValue> o2) {
-        return o1.value.key.compareTo(o2.value.key) & o1.value.value.compareTo(o2.value.value);
+        return o1.value.key.compareTo(o2.value.key) & compareValues(o1.value, o2.value);
     }
+
+    private int compareValues(CcdKeyValue o1, CcdKeyValue o2) {
+        if (o1.value == null && o2.value == null) {
+            return 0;
+        }
+        return o1.value.compareTo(o2.value);
```
[the remark this answers is not in the document]
Comment (aliveni) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java]: fixed it

File: src/integrationTest/resources/servicebus/message/new-application-example.json [hunk 4]
```diff
@@ -17,8 +17,14 @@
...
-    "field2": "value2s"
-  }
+  "ocr_data": [
+    {
+      "metadata_field_name": "field1",
+      "metadata_field_value": "value1"
+    },
+    {
+      "metadata_field_name": "field2",
+      "metadata_field_value": "value2"
```
Comment (rkondratowicz) [inline src/integrationTest/resources/servicebus/message/new-application-example.json]: used to be `value2s` (line 22), is this fine?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java]: just a suggestion here to make it more readable
```
if(o1.value == null && o2.value == null)
{
	return 0;
}


if (o1.value == null && o2.value != null) {

	return -1;

}

if (o1.value != null && o2.value == null) {

	return 1;
	
}

return o1.value.compareTo(o2.value);```

[the remark this answers is not in the document]
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java]: Also can we please add unit tests to cover all branches

File: src/integrationTest/resources/servicebus/message/new-application-example.json (hunk 4 above)
Comment (aliveni) [reply src/integrationTest/resources/servicebus/message/new-application-example.json]: I think this is fine. we are just verifying the data

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/CcdCollectionElementComparator.java]: realised this is only used for test so fine by me

**Pull request merged**